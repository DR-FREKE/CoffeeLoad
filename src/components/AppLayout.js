import React from 'react';
import {SafeAreaView, View, StyleSheet} from 'react-native';
import {Layout, TopNavigation} from '@ui-kitten/components';
import PropTypes from 'prop-types';

const AppLayout = ({children, ...props}) => (
  <Layout style={styles.scafold}>
    <SafeAreaView>
      {props.showTopNav && (
        <TopNavigation
          title={props.title}
          subtitle={props.subtitle}
          alignment={props.alignment}
          accessoryLeft={props.accessoryLeft}
          accessoryRight={props.accessoryRight}
          style={[
            styles.topNav,
            {backgroundColor: props.isThemeDark ? '#A47551' : 'transparent'},
          ]}
        />
      )}
      <View>{children}</View>
    </SafeAreaView>
  </Layout>
);

AppLayout.defaultProps = {
  alignment: 'start',
};

const styles = StyleSheet.create({
  scafold: {
    flex: 1,
    backgroundColor: '#D0B49F',
  },
  topNav: {
    // backgroundColor: 'transparent',
    paddingHorizontal: 20,
  },
});

export default AppLayout;
