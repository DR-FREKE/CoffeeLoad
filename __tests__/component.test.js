import React from 'react';
import AppLayout from '../src/components/AppLayout';
import renderer from 'react-test-renderer';

describe('every component should be rendered properly', () => {
  it('should render the AppLayout component', () => {
    const tree = renderer.create(<AppLayout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
