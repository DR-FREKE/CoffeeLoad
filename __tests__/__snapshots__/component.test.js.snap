// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`every component should be rendered properly should render the AppLayout component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View />
</RCTSafeAreaView>
`;
